### ISSUE TITLE
[SCREENREADER]: GIBCT® Colmery 107 - Zip code entry field does not announce valid city when a value is returned

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#1655]
## Description

The display for a profile detail includes a city when I enter a valid zipcode for `Other` class locations. This text is not announced to screen readers, and could leave users wondering if they'd entered data correctly. Screenshot attached below.

/cc @emilywaggoner for design consult, because this might be a visual adjustment, or a code adjustment.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

* As a screen reader user, I want to understand that I've entered a zip code correctly, and the UI has returned a valid city. I want to hear either the valid city + state read aloud, or the error message as a call to correct my entry.

## Environment

* MacOS Mojave
* Safari
* VoiceOver
* https://staging.va.gov/gi-bill-comparison-tool/profile/11806124

## Possible Fixes (optional)

I can see two improvements, with their own levels of complexity and effort:

1. We could use a typeahead that returns 1 or more suggestions as users are typing in zip codes.
2. We could add an `aria-live` region to the existing code, and some SR-only text. This second option could look like the following snippet. Content and right language would need to be decided upon.

```
<p aria-live="polite" aria-atomic="true">
  <span class="sr-only">Your postal code is located in
  <strong>Ludowici, GA</strong>
</p>
```

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2019-09-09 at 2.35.47 PM.png](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/91dabd43-b376-456b-aea9-d66d65d6cda9)

**Comment by sonntag-adam** [comment]
Re tested
##
![Screen Shot 2019-09-13 at 1.40.19 PM.png](https://images.zenhubusercontent.com/5c98df9c534e09000148f8d1/fb0cff5e-8827-45a0-8e81-9fb51083449f)
![Screen Shot 2019-09-13 at 1.41.21 PM.png](https://images.zenhubusercontent.com/5c98df9c534e09000148f8d1/b9ad7d89-eaa5-4ec8-8193-6c02c5c5135a)

**Comment by 1Copenut** [comment]
@ksantiagomorgan6 && @desireeturner I reviewed this one on Staging this morning. Results:

* MacOS + Safari + VoiceOver working as expected
* Win10 + IE11, Chrome + JAWS is reading the "Your postal code is located in..." too early
* Win10 + NVDA + Firefox is reading the "Your postal code is located in..." too early

I think this has to do with the text block "Your postal code is located in" being included when the `aria-live` region is added to the page. Right now we have two states, the state where the live region is added with "Your postal code is located in", and the state where it's either Loading or returning a result. 

Thinking in UX states, I believe we need three distinct conditions:
1. User selects `Other` from the dropdown. We add the `aria-live` region with no text inside it.
2. User starts typing into the Postal Code field. We add the SR-only text + "Loading". Screen readers should read "Your postal code is in loading dot dot dot".
3. We return a valid postal code and update the live region. Screen readers should read "Your postal code is in Grenada, MS."

LMK if you want to pair code or review this further.

**Comment by 1Copenut** [comment]
@ksantiagomorgan6 && @desireeturner I retested this with NVDA and JAWS today and it's working as expected. Moving to close.

I found one oddity with the error messaging if I entered a valid ZIP code with no rates found, but I'll open that as a separate issue.
